### PR TITLE
test: move to github actions

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

Move from travis to github actions

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

The last couple of months travis has become really slow to start so maybe time to try something new

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        golang:
+          - 1.14
+          - 1.15
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.golang }}
+      - uses: actions/cache@v2
```
Comment (threebee) [inline .github/workflows/pr.yml:34]: Can we please put names for each step? Currently, it looks confusing. Each step have `uses` but only one step have `name`. (except run tests)

Comment (gssbzn) [reply .github/workflows/pr.yml]: add to this but a name for `actions/checkout@v2` would be "checkout" which seems redundant

Comment (threebee) [reply .github/workflows/pr.yml]: For me checkout and cache (what?) were not very clear.

Step name suggestions:

- name `Checkout repository`, uses `actions/checkout@v2`
- name `Install Go`, uses `actions/setup-go@v2`
- name `Cache dependencies and build outputs`, uses `actions/cache@v2`

**Review by threebee — APPROVED** [review]
LGTM!

**Pull request merged**